// faye-websocket/lib/faye/websocket/api/event_target.js
/***
 * Excerpted from "CoffeeScript: Accelerated JavaScript Development, Second Edition",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/tbcoffee2 for more book information.
***/
var Event = require('./event');

var EventTarget = {
  onopen:     null,
  onmessage:  null,
  onerror:    null,
  onclose:    null,

  addEventListener: function(eventType, listener, useCapture) {
    this._listeners = this._listeners || {};
    var list = this._listeners[eventType] = this._listeners[eventType] || [];
    list.push(listener);
  },

  removeEventListener: function(eventType, listener, useCapture) {
    if (!this._listeners || !this._listeners[eventType]) return;

    if (!listener) {
      delete this._listeners[eventType];
      return;
    }
    var list = this._listeners[eventType],
        i    = list.length;

    while (i--) {
      if (listener !== list[i]) continue;
      list.splice(i,1);
    }
  },

  dispatchEvent: function(event) {
    event.target = event.currentTarget = this;
    event.eventPhase = Event.AT_TARGET;

    if (this['on' + event.type])
      this['on' + event.type](event);

    if (!this._listeners || !this._listeners[event.type]) return;

    this._listeners[event.type].forEach(function(listener) {
      listener(event);
    }, this);
  }
};

module.exports = EventTarget;
