// grunt-eco/Gruntfile.js
/***
 * Excerpted from "CoffeeScript: Accelerated JavaScript Development, Second Edition",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/tbcoffee2 for more book information.
***/
module.exports = function(grunt) {
  'use strict';

  // Project configuration
  grunt.initConfig({

    jshint: {
      options: {
        curly:   true,
        eqeqeq:  true,
        immed:   true,
        latedef: true,
        newcap:  true,
        noarg:   true,
        sub:     true,
        undef:   true,
        boss:    true,
        eqnull:  true,
        node:    true
      },

      gruntfile: 'Gruntfile.js',
      tasks: 'tasks/*.js',
      tests: '<%= nodeunit.tests %>'
    },

    // Prepare clean testing ENV
    clean: {
      test: ['tmp']
    },

    eco: {
      filesTest: {
        files: {
          'tmp/filesTest/all.js': ['tests/fixtures/**/*.eco'],
          'tmp/filesTest/flatOnly.js': 'tests/fixtures/*.eco'
        }
      },

      srcDestTest: {
        src: ['tests/fixtures/**/*.eco'],
        dest: 'tmp/srcDestTest/all.js'
      },

      expandTest: {
        files: [{
          expand: true,
          src: 'tests/fixtures/**/*.eco',
          dest: 'tmp/expandTest',
          ext: '.js'
        }]
      },

      basePathStringTest: {
        src: ['tests/fixtures/**/*.eco'],
        dest: 'tmp/basePathStringTest/all.js',
        options: {
          basePath: 'tests/fixtures'
        }
      },

      basePathFunctionTest: {
        src: ['tests/fixtures/**/*.eco'],
        dest: 'tmp/basePathFunctionTest/all.js',
        options: {
          basePath: function(jstPath){
            return jstPath.replace(/(^tests\/fixtures\/|-example$)/g, '');
          }
        }
      },

      amdTest: {
        options: {
          amd: true
        },
        files: [{
          expand: true,
          src: 'tests/fixtures/**/*.eco',
          dest: 'tmp/amdTest',
          ext: '.js'
        }]
      },

      noJstGlobalCheckTest: {
        options: {
          jstGlobalCheck: false
        },
        files: {
          'tmp/noJstGlobalCheckTest/all.js': 'tests/fixtures/**/*.eco'
        }
      }
    },

    nodeunit: {
      tests: ['tests/*_test.js']
    }
  });

  // Load local tasks
  grunt.loadTasks('tasks');

  // Load development tasks
  grunt.loadNpmTasks('grunt-contrib-clean');
  grunt.loadNpmTasks('grunt-contrib-jshint');
  grunt.loadNpmTasks('grunt-contrib-nodeunit');

  grunt.registerTask('default', ['jshint', 'test']);
  grunt.registerTask('test', ['clean', 'eco', 'nodeunit']);

};
