// nodemon/lib/config/defaults.js
/***
 * Excerpted from "CoffeeScript: Accelerated JavaScript Development, Second Edition",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/tbcoffee2 for more book information.
***/
// default options for config.options
module.exports = {
  restartable: 'rs',
  execMap: {
    py: 'python',
    rb: 'ruby',
    // more can be added here such as ls: lsc - but please ensure it's cross
    // compatible with linux, mac and windows, or make the default.js dynamically
    // append the `.cmd` for node based utilities
  },
  ignore: ['.git', 'node_modules', 'bower_components', '.sass-cache'],
  watch: ['*.*'],
  stdin: true,
  verbose: false,
  // 'stdout' refers to the default behaviour of a required nodemon's child,
  // but also includes stderr. If this is false, data is still dispatched via
  // nodemon.on('stdout/stderr')
  stdout: true

};
