// inquirer/lib/inquirer.js
/***
 * Excerpted from "CoffeeScript: Accelerated JavaScript Development, Second Edition",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/tbcoffee2 for more book information.
***/
/**
 * Inquirer.js
 * A collection of common interactive command line user interfaces.
 */

var inquirer = module.exports;


/**
 * Client interfaces
 */

inquirer.prompts = {
  list     : require("./prompts/list"),
  input    : require("./prompts/input"),
  confirm  : require("./prompts/confirm"),
  rawlist  : require("./prompts/rawlist"),
  expand   : require("./prompts/expand"),
  checkbox : require("./prompts/checkbox"),
  password : require("./prompts/password")
};

inquirer.Separator = require("./objects/separator");

inquirer.ui = {
  BottomBar: require("./ui/bottom-bar"),
  Prompt: require("./ui/prompt")
};


/**
 * Public CLI helper interface
 * @param  {Array}   questions  Questions settings array
 * @param  {Function} cb        Callback being passed the user answers
 * @return {null}
 */

inquirer.prompt = function( questions, allDone ) {
  return new inquirer.ui.Prompt( questions, allDone );
};
