// es5-ext/array/#/index.js
/***
 * Excerpted from "CoffeeScript: Accelerated JavaScript Development, Second Edition",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/tbcoffee2 for more book information.
***/
'use strict';

module.exports = {
	binarySearch:   require('./binary-search'),
	clear:          require('./clear'),
	compact:        require('./compact'),
	concat:         require('./concat'),
	contains:       require('./contains'),
	copyWithin:     require('./copy-within'),
	diff:           require('./diff'),
	eIndexOf:       require('./e-index-of'),
	eLastIndexOf:   require('./e-last-index-of'),
	entries:        require('./entries'),
	exclusion:      require('./exclusion'),
	fill:           require('./fill'),
	filter:         require('./filter'),
	find:           require('./find'),
	findIndex:      require('./find-index'),
	first:          require('./first'),
	firstIndex:     require('./first-index'),
	flatten:        require('./flatten'),
	forEachRight:   require('./for-each-right'),
	keys:           require('./keys'),
	group:          require('./group'),
	indexesOf:      require('./indexes-of'),
	intersection:   require('./intersection'),
	isCopy:         require('./is-copy'),
	isUniq:         require('./is-uniq'),
	last:           require('./last'),
	lastIndex:      require('./last-index'),
	map:            require('./map'),
	remove:         require('./remove'),
	separate:       require('./separate'),
	slice:          require('./slice'),
	someRight:      require('./some-right'),
	splice:         require('./splice'),
	uniq:           require('./uniq'),
	values:         require('./values')
};
module.exports[require('es6-symbol').iterator] = require('./@@iterator');
