// es5-ext/number/index.js
/***
 * Excerpted from "CoffeeScript: Accelerated JavaScript Development, Second Edition",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/tbcoffee2 for more book information.
***/
'use strict';

module.exports = {
	'#':              require('./#'),
	EPSILON:          require('./epsilon'),
	isFinite:         require('./is-finite'),
	isInteger:        require('./is-integer'),
	isNaN:            require('./is-nan'),
	isNumber:         require('./is-number'),
	isSafeInteger:    require('./is-safe-integer'),
	MAX_SAFE_INTEGER: require('./max-safe-integer'),
	MIN_SAFE_INTEGER: require('./min-safe-integer'),
	toInteger:        require('./to-integer'),
	toPosInteger:     require('./to-pos-integer'),
	toUint32:         require('./to-uint32')
};
