// es5-ext/object/index.js
/***
 * Excerpted from "CoffeeScript: Accelerated JavaScript Development, Second Edition",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/tbcoffee2 for more book information.
***/
'use strict';

module.exports = {
	assign:           require('./assign'),
	clear:            require('./clear'),
	compact:          require('./compact'),
	compare:          require('./compare'),
	copy:             require('./copy'),
	copyDeep:         require('./copy-deep'),
	count:            require('./count'),
	create:           require('./create'),
	eq:               require('./eq'),
	every:            require('./every'),
	filter:           require('./filter'),
	firstKey:         require('./first-key'),
	flatten:          require('./flatten'),
	forEach:          require('./for-each'),
	getPropertyNames: require('./get-property-names'),
	is:               require('./is'),
	isArrayLike:      require('./is-array-like'),
	isCallable:       require('./is-callable'),
	isCopy:           require('./is-copy'),
	isCopyDeep:       require('./is-copy-deep'),
	isEmpty:          require('./is-empty'),
	isObject:         require('./is-object'),
	isPlainObject:    require('./is-plain-object'),
	keyOf:            require('./key-of'),
	keys:             require('./keys'),
	map:              require('./map'),
	mapKeys:          require('./map-keys'),
	normalizeOptions: require('./normalize-options'),
	mixin:            require('./mixin'),
	mixinPrototypes:  require('./mixin-prototypes'),
	primitiveSet:     require('./primitive-set'),
	safeTraverse:     require('./safe-traverse'),
	setPrototypeOf:   require('./set-prototype-of'),
	some:             require('./some'),
	toArray:          require('./to-array'),
	validCallable:    require('./valid-callable'),
	validObject:      require('./valid-object'),
	validValue:       require('./valid-value')
};
