diff -Nur code/capybara/01/app/views/search/show.html.erb code/capybara/02/app/views/search/show.html.erb
--- code/capybara/01/app/views/search/show.html.erb	2011-05-25 21:32:48.000000000 +0100
+++ code/capybara/02/app/views/search/show.html.erb	2011-05-26 07:24:19.000000000 +0100
@@ -1 +1,6 @@
-Search page
+<form id="search" method="get" action="/search">
+  <fieldset>
+    <input type="text" id="query" name="query" />
+    <input type="submit" value="Search" />
+  </fieldset>
+</form>
