diff -Nur 10/app/models/message.rb 11/app/models/message.rb
--- 10/app/models/message.rb	2011-11-10 20:35:52.000000000 +0000
+++ 11/app/models/message.rb	2011-11-10 20:41:10.000000000 +0000
@@ -2,6 +2,6 @@
   belongs_to :user
 
   def self.like(content)
-    content.nil? ? [] : where(['content LIKE ?', "%#{content}%"])
+    content.nil? ? [] : where(['content LIKE ?', "%#{content}%"]).reverse
   end
 end
Binary files 10/db/test.sqlite3 and 11/db/test.sqlite3 differ
diff -Nur 10/features/support/screenshots.rb 11/features/support/screenshots.rb
--- 10/features/support/screenshots.rb	1970-01-01 01:00:00.000000000 +0100
+++ 11/features/support/screenshots.rb	2011-11-10 20:50:41.000000000 +0000
@@ -0,0 +1,6 @@
+After('@javascript') do |scenario|
+  if(scenario.failed?)
+    page.driver.browser.save_screenshot("html-report/#{scenario.__id__}.png")
+    embed("#{scenario.__id__}.png", "image/png", "SCREENSHOT")
+  end
+end
\ No newline at end of file
