// es5-ext/string/#/index.js
/***
 * Excerpted from "CoffeeScript: Accelerated JavaScript Development, Second Edition",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/tbcoffee2 for more book information.
***/
'use strict';

module.exports = {
	at:                     require('./at'),
	camelToHyphen:          require('./camel-to-hyphen'),
	capitalize:             require('./capitalize'),
	caseInsensitiveCompare: require('./case-insensitive-compare'),
	codePointAt:            require('./code-point-at'),
	contains:               require('./contains'),
	hyphenToCamel:          require('./hyphen-to-camel'),
	endsWith:               require('./ends-with'),
	indent:                 require('./indent'),
	last:                   require('./last'),
	normalize:              require('./normalize'),
	pad:                    require('./pad'),
	plainReplace:           require('./plain-replace'),
	plainReplaceAll:        require('./plain-replace-all'),
	repeat:                 require('./repeat'),
	startsWith:             require('./starts-with')
};
module.exports[require('es6-symbol').iterator] = require('./@@iterator');
